### ISSUE TITLE
Restore Build Scan Capability

### ISSUE THREAD

**Issue by jzheaux** [spring-projects/spring-security#15120]
For the `5.8.x` branch, [using the `--scan` attribute does not work](https://docs.gradle.com/develocity/gradle-plugin/current/#publishing_on_demand) per the following from Gradle:

> Gradle versions older than 8.4 do not recognize the Develocity plugin as the replacement for deprecated Gradle Enterprise and Build Scan plugins. When --scan argument is used, Gradle attempts to apply its version of the plugin which may conflict with the Develocity plugin explicitly applied to your project. If the plugin applied by Gradle wins, the new functionality provided by the Develocity plugin will not be available.

As such, `--scan` needs to be removed for the moment and likely replaced with [a recommendation like the following](https://docs.gradle.com/develocity/gradle-plugin/current/#publishing_based_on_criteria):

```groovy
develocity {
    buildScan {
        if (System.getenv("IS_A_PULL_REQUEST")) {
            publishing.onlyIf { true }
            server = "https://scans.gradle.com"
        }
    }
}
```

**Comment by jzheaux** [comment]
The above did not quite work since the Spring Gradle Enterprise Conventions plugin needs to set the `server` value and the Develocity plugin does not allow it to be overridden.

**Comment by jzheaux** [comment]
Another way to do this is to only apply the Spring Gradle Enterprise Conventions plugin when enterprise credentials are present:

```groovy
if (System.getenv("DEVELOCITY_ACCESS_KEY")) {
	apply plugin: "io.spring.ge.conventions"
}
```

This works because the `DEVELOCITY_ACCESS_KEY` credential is not available to PRs. GIven that, it can't exercise the Conventions feature set anyway.

**Comment by jzheaux** [comment]
Closed in [ca67634](https://github.com/spring-projects/spring-security/pull/15152/commits/ca67634110eaf52f025eeff9ad448a6ca8025002)
